http-core/src/test/scala/org/apache/pekko/http/impl/model/parser/HttpHeaderSpec.scala

```diff
@@ -876,6 +876,13 @@ class HttpHeaderSpec extends AnyFreeSpec with Matchers {
       parse("User-Agent", "(" * 10000).errors.head shouldEqual
       ErrorInfo("Illegal HTTP header 'User-Agent': Illegal header value", "Header comment nested too deeply")
     }
+
+    "should not broken when header-value is null" in {
+      val errors = parse("Content-Disposition", null).errors
+      errors.size shouldBe 1
```

Comment:
`errors should have size 1` is a better way to do this. When this fails, scalatest will say something like 
`List(a1, a2) did not have size 1`

With `errors.size shouldBe 1` fails, you will get an error like `2 was not 1`. I much prefer getting the detailed message from my assertion. I would strongly recommend that you read up on scalatest. The assertions are carefully designed to give us lots of context when assertions fail. But if you go off and use the simplest assertions then you get very little info when things fail.

Reply:
done